### ISSUE TITLE
%sveltekit.assets% should behave the same for "shortcut icon" and "apple-touch-icon" as it does for "icon"

### ISSUE THREAD

**Issue by schlundd** [sveltejs/kit#10317]
### Describe the problem

I am getting lots of 404 errors for apple-touch-icon files as (some?) browsers try to load it from deeper paths.

### Describe the proposed solution

This was solved for <link rel="icon"> in https://github.com/sveltejs/kit/pull/5583.
The same behaviour should be applied to "shortcut icon" and "apple-touch-icon"

### Importance

nice to have

**Comment by dummdidumm** [comment]
Whoever implements this should have access to a Safari browser to check that the weird scroll breakage observed in an earlier version of #5583 does not occur.

**Comment by jasongitmail** [comment]
FWIW, SvelteKit automatically adding the origin to create absolute URLs for certain images, but not others, was a _huge headache_ getting started with Svelte Kit for a couple days as I figured out how to handle metadata for my images and when to include the origin for canonical URLs, icons, og image, etc.

It was difficult because `npm run dev` outputs relative URLs (makes sense because of Vite), `npm run build` outputs relative URLs (okay good to know), but running the production app has absolute URLs  in dev console (!) for some paths added automatically but not all(!) and inspecting the actual source (cmd+option+u in Chrome) shows relative paths in production.

I ended up with double origins in my URLs before realizing SK adds them automatically. 

TLDR I'd recommend not making this permanent in SK. It's worth revisiting if the browser bug warranting this has been resolved because the fix makes SK's DX unpredictable for URL handling. Or at the very least, should be documented. 

IMO, for a real project, it doesn't make sense to include any assets, other than `favicon.ico` directly from the static dir, which might make this issue void.

It's better to include them in a `<svelte:head> ... </svelte:head>` in a main +layout.svelte so they can be hashed and benefit from far futures cache headers, which may help address any browser 404 issue. Only `favicon.ico` should be unhashed.

Here's a really well-researched article on the required meta for favicon and touch icons for today's browsers: https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs

> I am getting lots of 404 errors for apple-touch-icon files as (some?) browsers try to load it from deeper paths.

Curious if your paths start with `/` like `/image.png` or are `image.png`? The latter could cause issues.

**Comment by Serator** [comment]
I'm using the same method described by @jasongitmail, because I've encountered a problem that when posting a message to Slack, Slack uses `apple-touch-icon` for the site icon in the message preview and can't get it from `<link rel="apple-touch-icon" href="%sveltekit.assets%/apple-touch-icon.png" />` from `src/app.html`.

P.S. The article from EvilMartians is really great, but partially outdated. Chrome, due to a [bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1450857&sort=-status&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified&x=m&y=releaseblock&cells=tiles&q=is%3Astarred%20favicon&can=1), loads both the ICO and SVG version of the favicon. You can check this in Chrome at the link from the previous post. On my side, I fixed this by replacing `sizes="any"` with `sizes="32x32"`.

**Comment by ai** [comment]
@Serator thanks for the note, I updated [the article](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs)

**Comment by andylib93** [comment]
@ai **loved** your article!! Just a little update:
In chapter **Step 2: Create an ICO file** the cli argument for inkscape is (maybe nowadays) invalid. I made it work with `--export-png` instead of `--export-filename` :blush:

Accordingly for chapter **Step 3: Create PNG images** :+1:

**Comment by ai** [comment]
@andylib93 thanks =^_^=

Strange.

```
$ inkscape --help | grep export-filename
  -o, --export-filename=FILENAME             Output file name (defaults to input filename; file type is guessed from extension if present; use '-' to write to stdout)
	inkscape --export-filename=out.pdf in.svg
```

And I can’t find `export-png` in `inkscape --help`.

**Comment by andylib93** [comment]
@ai that is indeed strange, but it seems like I need to upgrade my version, in the [cli-documentation](https://wiki.inkscape.org/wiki/Using_the_Command_Line) under **Deprecations and Replacements** it says `--export-png` needs to be replaced with `--export-type="xyz"`.

My mistake 😇
